File: spring-modulith-with-ddd/src/main/java/com/example/springmodulithwithddd/inventory/BookRepository.java
package com.example.springmodulithwithddd.inventory;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository //visibility of this repository interface is package-private and not public
interface BookRepository extends JpaRepository<Book, Long> {
    Optional<Book> findByIsbn(String isbn);

    Optional<Book> findByInventoryNumber(Book.Barcode inventoryNumber);

    List<Book> findByStatus(Book.BookStatus status);
}

File: data-rest-kotlin/src/main/kotlin/com/example/datarestkotlin/BookRepository.kt
package com.example.datarestkotlin

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface BookRepository : PagingAndSortingRepository<Book, Long> {
    fun findByTitleContaining(@Param("query") query: String, page: Pageable): Page<Book>
    fun findByAuthorContaining(@Param("query") query: String, page: Pageable): Page<Book>
}